### ISSUE TITLE
Duplicate webhook event

### ISSUE THREAD

**Issue by buonzz** [reapit/foundations#7324]
**Describe the bug**
tenancy.created webhook should be only fired once, not multiple times in a very short period of time.

**To Reproduce**

1. Go to https://developers.reapit.cloud/
2. Go to API > Webhooks 
3. Select an App. Click Manage Webhooks
4. add Webhook URL, Subscription topics "Tenany Created"

**Expected behaviour**
when tenancy is created for entity GRE220061, i expect it to make a POST request to my webhook only once, but it did twice:
tenancy.created webhook fired at 2022-07-14T12:50:31.04+00:00
then another at 2022-07-14T12:50:32.911+00:00

**Screenshots**
<img width="1002" alt="Screen Shot 2022-07-18 at 8 51 19 AM" src="https://user-images.githubusercontent.com/946233/179679331-56115371-b842-4fea-bbad-bbdd565a94a0.png">

**Additional context**
when i diff the contents of the two payloads, there isn't really something that differed aside from event id and the timestamp
<img width="933" alt="Screen Shot 2022-07-18 at 9 04 15 AM" src="https://user-images.githubusercontent.com/946233/179679520-f5ec71e6-6179-41dc-bbff-b121d50d0696.png">

Here is the json files for those payloads we have received.
[GRE220061-payloads.zip](https://github.com/reapit/foundations/files/9138040/GRE220061-payloads.zip)

**Specification**
- Investigate the cause of duplicate webhook events
- Speak with @plittlewood-rpt about the occurrence of this

**Comment by buonzz** [comment]
im setting it was well in "Selling property exchanged"

<img width="998" alt="Screen Shot 2022-07-21 at 12 05 35 PM" src="https://user-images.githubusercontent.com/946233/180127869-5f930fb6-fd4a-41c1-a222-540a6f89b402.png">

**1st event**

```
topic: properties.selling.exchanged
eventId: 98dfe74b-a5d8-4054-8358-e8aaa8c6eb12
eventTime: 2022-07-20T14:48:41.8378857Z
entityId: LEE210317
customerId: CNS
```

**2nd event**

```
topic: properties.selling.exchanged
eventId: ce09b542-0ab8-4075-8eaf-0d7a7e7d2b84
eventTime: 2022-07-20T14:48:41.8362271Z
entityId: LEE210317
customerId: CNS
```

**Comment by plittlewood-rpt** [comment]
Hi @buonzz - has this happened just on one occasion or is this a persistent issue?

**Comment by buonzz** [comment]
hi, its persistent issue. I'm seeing multiple entries we received in our logs. The impact of it in our integration are:

* causing unpredictable race-conditions, wherein two webhook is trying to manipulate same data at the same time.
* we are being billed unnecessarily for api requests that is redundant

Thanks

**Comment by plittlewood-rpt** [comment]
Thanks @buonzz  - is it just this single customer id that you're seeing this for? Or are there others?

**Comment by buonzz** [comment]
we only have one customer actively using our app atm (CNS), so cant really say if it happens to multiple customers

**Comment by buonzz** [comment]
Hi,

its intermittent.

I did look at the configuration of webhooks, and there is only one active webhook at the moment.
It also doesnt explain the different eventId values:

1st event
```
topic: properties.selling.exchanged
eventId: 98dfe74b-a5d8-4054-8358-e8aaa8c6eb12
eventTime: 2022-07-20T14:48:41.8378857Z
entityId: LEE210317
customerId: CNS
```

2nd event

```
topic: properties.selling.exchanged
eventId: ce09b542-0ab8-4075-8eaf-0d7a7e7d2b84
eventTime: 2022-07-20T14:48:41.8362271Z
entityId: LEE210317
customerId: CNS
```

for example following is two `properties.selling.exchanged` events with eventId `98dfe74b-a5d8-4054-8358-e8aaa8c6eb12` and `ce09b542-0ab8-4075-8eaf-0d7a7e7d2b84`
If they are the same event, posted twice due to configuration their eventId would be  different.

and they also show up in the developer portal itself as duplicate (e.g. this is not even our log files):
<img width="998" alt="180127869-5f930fb6-fd4a-41c1-a222-540a6f89b402" src="https://user-images.githubusercontent.com/946233/181877906-2a2942e0-283e-451b-9ba7-43f59219ca0c.png">

This seems started to show up around July 14 onwards:

https://developers.reapit.cloud/webhooks/logs?applicationId=63bbe303-4ced-4f2f-a83c-35a870d2af7a&from=2022-07-13&to=2022-07-14

GRE220060
<img width="1219" alt="Screen Shot 2022-07-30 at 2 36 17 PM" src="https://user-images.githubusercontent.com/946233/181878099-9a16b8eb-1265-4b9a-ab53-fe758c362b7e.png">


GRE220061
<img width="1254" alt="Screen Shot 2022-07-30 at 2 37 54 PM" src="https://user-images.githubusercontent.com/946233/181878133-946f078d-e351-4d90-86c6-808ce899ea93.png">

BRO220078
<img width="1246" alt="Screen Shot 2022-07-30 at 2 38 49 PM" src="https://user-images.githubusercontent.com/946233/181878173-b608b451-f571-4b4f-a26c-1aef84365420.png">

weird enough CHA220040 only fired once:
<img width="1198" alt="Screen Shot 2022-07-30 at 2 39 55 PM" src="https://user-images.githubusercontent.com/946233/181878212-4623f67f-30ec-4caf-82ca-60c2547c5f65.png">

despite being on the same day as others.

I've checked for the July 29th


GRE211570 - 3 events for Selling property exchanged
GRE220017 - 2 events for Selling property exchanged
<img width="1243" alt="Screen Shot 2022-07-30 at 2 42 41 PM" src="https://user-images.githubusercontent.com/946233/181878448-cabb13a0-4ff9-4209-a389-c1f1f40b7653.png">

note that all samples/screenshot above is taken from https://developers.reapit.cloud/ itself and not even our logs. So I could not think of any configuration change in our end to fix the issue.

**Comment by plittlewood-rpt** [comment]
Hi @buonzz - I've added some more logging to try and catch where this is originating from. Can you let me know the next time you encounter one of these please (from now 01-08-2022 16:11 BST onwards)

**Comment by buonzz** [comment]
hi,

there is one for GRE220070

<img width="1179" alt="Screen Shot 2022-08-03 at 7 21 40 AM" src="https://user-images.githubusercontent.com/946233/182492832-0c791360-3dc8-44c4-91c8-91a5dce23367.png">

1st event
```
timestamp: 2022-08-02T10:51:12.983+00:00
eventId: 830472a6-32f2-495c-b2ec-373808787d10
```

2nd event:
```
timestamp: 2022-08-02T10:51:13.72+00:00
eventId: 5490436c-9aad-411d-a803-d8cc10af7169
```

thanks

**Comment by plittlewood-rpt** [comment]
Thanks - this is useful I think I can see what's happened here. I should now be able to replicate and work on a suitable resolution

**Comment by plittlewood-rpt** [comment]
@buonzz I'll shortly be pushing out an update to hopefully resolve the issue you are seeing where you get multiple tenancy created hooks in some scenarios. This stems from the tenancy being created in a 'current' state rather than being added as 'arranging' and then moving to 'current' - as the documentation mentions, we do not guarantee you'll only get each event once. This is a bit of an edge case which will hopefully be under control with the update going out shortly.

I have looked at your other example for property GRE220017 where you got multiple Selling property exchanged hooks. This looks to have been caused by the customer's use of the system, where they have changed the status of the property from Exchanged back to Sold STC and back again multiple times. The action of the property status being changed to Exchanged is what generates the "Selling property exchanged" hook so this looks like expected behaviour.
